cantools/database/diagnostics/data.py

```diff
@@ -41,10 +44,10 @@ def __init__(self,
         self.byte_order: ByteOrder = byte_order
 
         #: The minimum value of the data, or ``None`` if unavailable.
-        self.minimum: Optional[float] = minimum
+        self.minimum: Optional[Union[float, List[float]]] = minimum
 
         #: The maximum value of the data, or ``None`` if unavailable.
-        self.maximum: Optional[float] = maximum
+        self.maximum: Optional[Union[float, List[float]]] = maximum
```

Comment:
My understanding was that ordinary linear scaling would just be piecewise linear with only one "piece". Do you think we might run into issues handling it like that?

Generally, when the CDD is created/edited with CANdelaStudio it will be impossible to save the data type if the sample points are not monotonously increasing. Still might be a nice consistency check for the few people courageous enough to edit CDDs by hand :) I'll give it a go and see how much impact it'll have on the performance

Reply:
> My understanding was that ordinary linear scaling would just be piecewise linear with only one "piece". Do you think we might run into issues handling it like that?

AFAICS it will require changes to some code. within the cantools codebase itself, this is not a big deal as we can fix these issues. the problem is that there are probably quite a few affected downstream users whose code we cannot fix ourselves.

> Generally, when the CDD is created/edited with CANdelaStudio it will be impossible to save the data type if the sample points are not monotonously increasing.

I suppose that it also does not allow gaps or overlaps between the segments, i.e., it probably defines the segments as a list of raw values where the minimum of one segment is the maximum of the previous one instead of a list of [minimum, maximum] raw values?

> Still might be a nice consistency check for the few people courageous enough to edit CDDs by hand :) I'll give it a go and see how much impact it'll have on the performance

+1. that was my point ;)